src/source/vector_tile_source.js

```diff
@@ -104,10 +106,15 @@ class VectorTileSource extends Evented implements Source {
     load() {
         this._loaded = false;
         this.fire(new Event('dataloading', {dataType: 'source'}));
-        this._tileJSONRequest = loadTileJSON(this._options, this.map._requestManager, (err, tileJSON) => {
+        const language = this.map._language;
+        const worldview = this.map._worldview;
+        this._tileJSONRequest = loadTileJSON(this._options, this.map._requestManager, language, worldview, (err, tileJSON) => {
```

Comment:
```suggestion
        try {
                Intl.getCanonicalLocales(language);
        } catch (_) {
                console.warn(`Ensure that your requested language string is a valid BCP-47 code. Found: ${language}`);
        }
        
        if (worldview && worldview.length !== 2) console.warn(`Requested worldview strings must be a valid ISO alpha-2 code. Found: ${worldview}`);
        
        this._tileJSONRequest = loadTileJSON(this._options, this.map._requestManager, language, worldview, (err, tileJSON) => {
```

Reply:
Intl.getCanonicalLocales doesn't seem to do exactly what we're looking for. If you do `Intl.getCanonicalLocales('hello')` for example, it just returns the string `hello` which leads to a 400 error. I still think the best way to handle this is probably to surface the error so the developer can handle it.